schemas/content/product.schema.json

```diff
@@ -60,111 +59,50 @@
-        },
-        "xdm:unitOfMeasure": {
-          "title": "Unit of Measure",
-          "type": "string",
+            "The list of prices for this product."
+        },
+        "xdm:assets": {
+          "title": "Assets",
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "$ref":
+              "https://ns.adobe.com/xdm/assets/asset#/definitions/asset"
+          },
           "description":
-            "Standard unit of measure of the variant. Denotes the units for the size measurement."
+            "List of assets associated with this product. The primary product asset has to be the first list element."
         },
-        "xdm:countryOfOrigin": {
-          "title": "Country of Origin",
-          "type": "string",
-          "pattern": "^[A-Z]{2}$",
+        "xdm:attributes": {
+          "title": "Attributes",
+          "$ref":
+              "https://ns.adobe.com/xdm/content/product-attributes",
           "description":
-            "The two-character [ISO 3166-1 alpha-2](https://datahub.io/core/country-list) code for the country of origin of the product as defined by customs requirements."
-        },
-        "xdm:COGS": {
-          "title": "Cost of Good Sold",
-          "type": "number",
-          "description": "Cost of Good Sold. In the `currencyCode` currency."
-        },
-        "xdm:listPrice": {
-          "title": "List Price",
-          "type": "number",
-          "description":
-            "Default price of the product before sales and discounting. In the `currencyCode` currency."
-        },
-        "xdm:currencyCode": {
-          "title": "Currency Code",
-          "type": "string",
-          "examples": ["USD", "EUR"],
-          "pattern": "^[A-Z]{3}$",
-          "description":
-            "The ISO 4217 alphabetic currency code used for cost and pricing, including `listPrice` and `COGS`."
-        },
-        "xdm:originalSaleDate": {
-          "title": "Original Sale Date",
-          "type": "string",
-          "format": "date",
-          "description":
-            "First date the product was made available for sale. The time using RFC3339 with a stated timezone offset such as \"2001-07-04T12:08:56-07:00\". An example formatting pattern is \"yyyy-MM-dd'T'HH:mm:ssXXX\"."
-        },
-        "xdm:productCreateDate": {
-          "title": "Product Creation Date",
-          "format": "date",
-          "type": "string",
-          "description":
-            "The date when this product variant was created. The time using RFC3339 with a stated timezone offset such as \"2001-07-04T12:08:56-07:00\". An example formatting pattern is \"yyyy-MM-dd'T'HH:mm:ssXXX\"."
-        },
-        "xdm:productLastModified": {
-          "title": "Product Last Modified",
-          "format": "date",
-          "type": "string",
+            "Descriptive attributes associated with this product. "
+        },
+        "xdm:variants": {
```

Comment:
This creates a highly nested structure, which is not what I think we want to do. What we have here now is a single product master with multiple variants. This will be expensive to query as it is a table inside a table where almost all joins will be to the inner table. This is a very big issue.

Reply:
We will have more than one implementation of XDM, so I think there are bigger concerns than the query efficiency of one single implementation. In addition, almost any PIM implements some relationship of product masters to variants, so this is a join that is quite common and not enough rationale for me to reject the PR.

Reply:
@trieloff - I disagree. The current design is not designed to support the primary use case we see for Products which is that of an experience in the context of a product. Int his case the product being experienced is almost always a product variant. So it does not makes sense to me for this to be embedded in the master, which is more of a seldom used formality that is used for PIM data designs. 

To make an analogy, this is like building an CDN, but the only records it has is of templates with an inner list of the actual media that is being requested.

The previous design had variants as first class records and still maintained the concept of master records. While there are many improvements in this PR, this is not one and needs to be revisited.

Reply:
@cdegroot-adobe can you elaborate on
> The current design is not designed to support the primary use case we see for Products which is that of an experience in the context of a product.

I think there are different "we" views. We (the AEM commerce team) look into broader commerce, not only the client experience. For example I can think of importing PIM catalog data into ACP and running ML task on that data.

I think with the current proposal you can realize all kinds of product views. From experience context (the product schema has one needs, there is no need to use the nested variant structure) up to complex PIM data structures.